src/library/dao/genredao.cpp

```diff
@@ -0,0 +1,197 @@
+#include "library/dao/genredao.h"
+
+#include <QSqlError>
+#include <QSqlQuery>
+#include <QVariant>
+
+#include "library/dao/trackschema.h"
+#include "util/assert.h"
+#include "util/logger.h"
+
+namespace {
+mixxx::Logger kLogger("GenreDAO");
+
+#define LOG_FAILED_QUERY(query) qDebug() << __FILE__ << __LINE__ << "FAILED QUERY [" \
+                                         << (query).executedQuery() << "]" << (query).lastError()
+}
+
+GenreDAO::GenreDAO()
+        : DAO() {
+}
+
+DbId GenreDAO::addGenre(const QString& name) {
+    const QString trimmedName = name.trimmed();
+    if (trimmedName.isEmpty()) {
+        return DbId();
+    }
+
+    // First Check if genre exists
+    Genre existingGenre = getGenreByName(trimmedName);
+    if (existingGenre.isValid()) {
```

Comment:
Isn't this always valid? If we later re-decide about what valid means we would introduce a duplicate which is undesired. I think we should remove the check here.